src/Grassmann.jl

```diff
@@ -0,0 +1,275 @@
+using LinearAlgebra: svd, qr, diag, Diagonal, det
+import LinearAlgebra: norm
+@doc doc"""
+    Grassmann{n,k,T} <: Manifold
+
+The Grassmann manifold $\operatorname{Gr}(n,k)$ consists of all subspaces span
+by $k$ linear independent vectors $\mathbb F^n$, where $\mathbb F \in \{\mathbb R, \mathbb C\}$ is
+either the real- (or complex-) valued vectors. This yields all $k$-dimensional
+subspaces of $\mathbb R^n$ for the real-valued case and all $2k$-dimensional
+subspaces of $\mathbb C^n$ for the second.
+
+The manifold can be represented as
+````math
+\operatorname{Gr}(n,k) \coloneqq \bigl\{ \operatorname{span}(x)
+: x \in \mathbb F^{n\times k}, \bar{x}^\mathrm{T}x = I_k\},
+````
+where ${\bar\cdot}^{\mathrm{T}}$ denotes the complex conjugate transposed and
+$I_k$ is the $k\times k$ identity matrix. This means, that the columns of $x$
+form an orthonormal basis of the subspace, that is a point on
+$\operatorname{Gr}(n,k)$, and hence the subspace can actually be represented by
+a whole equivalence class of representers.
+Another interpretation is, that
+````math
+\operatorname{Gr}(n,k) = \operatorname{St}(n,k) / \operatorname{O}(k),
+````
+i.e the Grassmann manifold is the quotient of the [`Stiefel`](@ref) manifold and
+the orthogonal group $\operatorname{O}(k)$ of orthogonal $k\times k$ matrices.
+
+The tangent space at a point (subspace) $x$ is given by
+````math
+T_x\mathrm{Gr}(n,k) = \bigl\{
+ v \in \mathbb{F}^{n\times k} : 
+ {\bar v}^{\mathrm{T}}x + {\bar x}^{\mathrm{T}}v = 0_{k} \bigr\},
+````
+where $0_{k}$ denotes the $k\times k$ zero matrix.
```

Comment:
It might be worth pointing out that different representations of the same point on Grassmanian lead to different tangent spaces.

Reply:
well, they lead to different _representations_ of one and the same tangent space – but yes that could be added, you're right.

Reply:
Yes, that's what I mean -- different representations.